datafusion/src/scalar.rs

```diff
@@ -333,6 +333,29 @@ impl std::hash::Hash for ScalarValue {
     }
 }
 
+impl num::ToPrimitive for ScalarValue {
+    fn to_i64(&self) -> Option<i64> {
+        use ScalarValue::*;
+        match self {
+            Int8(v) => Some(v.unwrap() as i64),
```

Comment:
I think you can avoid the unwrap using something like:

```suggestion
            Int8(v) => v.map(|v| v.into()),
```